components/brave_origin/brave_origin_handler.h

```diff
@@ -0,0 +1,44 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_BRAVE_ORIGIN_BRAVE_ORIGIN_HANDLER_H_
+#define BRAVE_COMPONENTS_BRAVE_ORIGIN_BRAVE_ORIGIN_HANDLER_H_
+
+#include "brave/components/brave_origin/common/mojom/brave_origin.mojom.h"
+#include "mojo/public/cpp/bindings/pending_receiver.h"
+#include "mojo/public/cpp/bindings/receiver.h"
+
+namespace brave_origin {
+class BraveOriginService;
+
+class BraveOriginHandler : public mojom::BraveOriginHandler {
```

Comment:
`mojom::BraveOriginSettingsHandler` and class name `BraveOriginSettingsHandlerImpl`